avro-fastserde/src/main/java/com/linkedin/avro/fastserde/primitive/PrimitiveArrayList.java

```diff
@@ -0,0 +1,137 @@
+package com.linkedin.avro.fastserde.primitive;
+
+import java.util.AbstractList;
+import java.util.Iterator;
+import org.apache.avro.generic.GenericArray;
+import org.apache.avro.generic.GenericContainer;
+import org.apache.avro.generic.GenericData;
+
+
+public abstract class PrimitiveArrayList<T, L> extends AbstractList<T>
+    implements GenericContainer, Comparable<GenericArray<T>> {
+  private final Class<L> type;
+  protected int size = 0;
+
+  public PrimitiveArrayList(Class<L> type) {
```

Comment:
safety concern - whats preventing me from insta ntiating this with an argument that isnt a primitive?

Reply:
would it be safer to make this protected?